src/raid.cpp

```diff
@@ -292,13 +288,13 @@ void RaidBufferManager::submitBuffer(unsigned connectionNum, FilePiece* piece)
     {
         finalize(*piece);
         assert(asyncoutputbuffers.find(connectionNum) == asyncoutputbuffers.end() || !asyncoutputbuffers[connectionNum]);
-        asyncoutputbuffers[connectionNum] = piece;
+        asyncoutputbuffers[connectionNum].reset(piece);
     }
 }
 
-RaidBufferManager::FilePiece* RaidBufferManager::getAsyncOutputBufferPointer(unsigned connectionNum)
+std::shared_ptr<RaidBufferManager::FilePiece> RaidBufferManager::getAsyncOutputBufferPointer(unsigned connectionNum)
 {
-    std::map<unsigned, FilePiece*>::iterator i = asyncoutputbuffers.find(connectionNum);
+    auto i = asyncoutputbuffers.find(connectionNum);
```

Comment:
No need to add a commit for this `auto`s, but in cases like this one it would be better to keep the full declaration, so the reader knows at first sight what's in the first/second

Reply:
yes I've been on both sides of this, having both enjoyed the convenience of auto when writing code, and having thought it would be nice to know the exact types when reviewing.    This one isn't too long but some iterator names are ridiculous and auto is certainly better then.   In this case we can see that `find` is looking for a `connectionNum` so we know what the key of the map is (which corresponds to `first` in the pair of course.    And then the name of the map tells us what the value (ie `second`) is in this case - an output buffer (though without the exact type name of course).   In a good GUI you can just mouse over it to get the exact types of course.  Anyway, happy to keep it in mind and good to discuss.  cheers